config/routes.rb

```diff
@@ -4,7 +4,13 @@
   # Defines the root path route ("/")
   # root "articles#index"
   root 'users#index'
-  resources :users, only: [:index,:show] do
-    resources :posts, only: [:index,:show]
-end
+  get '/users', to: 'users#index', as: :users
+  get '/users:id', to: 'users#show', as: :user
+  get 'users:user_id/posts', to: 'posts#index', as: :user_posts
+  get 'users:user_id/posts:id', to: 'posts#show', as: :user_post
+  get '/posts/new', to: 'posts#new', as: :new_user_post
+  post '/posts/new', to: 'posts#create' #create_new_post
+  post 'comments/new', to: 'comments#create', as: :new_user_comment #create_new_comment
+  post 'likes/new', to: 'likes#create', as: :new_user_like #create_new_like
+  get 'users/:user_id/post/:id' => 'comments#new', as: 'user_comment'
```

Comment:
[OPTONAL] There's a much simpler way to define your routes. Rails will automatically create the paths for you.

```
  root "users#index"
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create]
    end
  end
```

- When you run `rails routes` you will see all the paths for each page

![image](https://user-images.githubusercontent.com/38088818/232190873-cb29d917-f755-413d-a480-ac424728b3e3.png)

### How the routes work

- `user_posts` - Gets all posts using path `/users/:user_id/posts`. This means it requires the `user_id`  as a parameter. 

For example:
```
      <%= link_to user_posts_path(@user) do %>
           ....
      <% end %>
````

   Notice how you simply append `path` to the `user_posts` route from the `rails routes` list to have `user_posts_path`.  The `user_posts_path` is like a routing method that accepts the required IDs as parameters. In this case, we only require the `user_id` to route to posts thus `user_posts_path(@user)`. This is how you use all the routes that are created by Rails using the `resources` keyword in the routes config file.

- `new_user_post` - Creates a new post in the `new` page using path `/users/:user_id/posts/new`. This means it requires the `user_id`  as a parameter. 

For example:
```
      <%= link_to new_user_post_path(@user) %>
````

- `user_post` - Shows one post in the post `show` page using path `/users/:user_id/posts/:id`. This means it requires the `user_id` and post `id`  as parameters. 

For example:
```
      <%= link_to user_post_path(@user, @post) %>
````

- `user_post_comments` - Gets all post comments using path `/users/:user_id/posts/:post_id/comments`. This means it requires the `user_id` and `post_id` as parameters. 
-  `new_user_post_comment` - Creates new comment for the `new` page using path `/users/:user_id/posts/:post_id/comments/new`. This means it requires the `user_id` and `post_id` as parameters. 

This way makes it easy for you to use all the defined routes without having to type them manually.